ggu/main.c

```diff
@@ -171,6 +210,36 @@ ggu_policy_iterator(struct ggu_decision *ggu_decision,
 				sizeof(policy.flow.f.v6));
 			params_offset = sizeof(policy.flow.f.v6);
 			break;
+		case GGU_DEC_IPV4_FLUSHED:
```

Comment:
Trying to implement a flush request as part of policy decisions is complicating the code because a flush request does not align well with a policy decision. This lack of alignment is rooted in the fact that a flush request doesn't have a flow id (i.e. source and destination address). For example, it's the lack of a flow id that forces us to send flush requests to all GK blocks.

My suggestion is to create a new packet format for flush requests. These requests will be rare and always generated by the explicit request of network operators to deal with some impactful change. We should even have a tool to send these requests out. I recommend using the name `gkadm` for this tool.

The format of these packets should be simple: just a list of IPv4 and IPv6 prefixes.

Reply:
Should we implement this ``gkadm`` tool in this pull request? If so, should ``gkadm`` take a file which contains a list of IPv4 and IPv6 prefixes as input?

Reply:
It's fine to have `gkadm` in another pull request; it will simplify the review of both pull requests. But if you do so, please add an issue to the first deployment milestone, so we have a reference to this missing part.

The very first version of `gkadm`, which we expect here, should just receive all tis parameters from the command line and send out the packet. Thus, `gkadm` doesn't need **a list** of prefixes for now.

Reply:
There are two ways that I came up to create a new packet format for the flush requests: (1) reusing the GGU header, i.e., ``struct ggu_common_hdr``. In this way, we will add a ``uint8_t decision_type`` field to distinguish between the policies from Grantor and administrators; (2) add a new pair of UDP port numbers for flush requests so that we can define a completely new packet format for the flush requests. 

Moreover, as we are allowing the administrators to explicitly flush the flows and separating the flush requests from the GGU policies, is it possible to implement this functionality in the dynamic configuration as Cody suggested before? Otherwise, the IP address of the Gatekeeper server and the MAC address of the nexthop should also be parameters of the ``gkadm`` program.

Let me know your thoughts.

Reply:
You are right, implementing the flush through the dynamic configuration block would be the easiest way for now. In fact, this even deal with the fact that these flushes are sensitive to security and that our GT-GK channel is not encrypted. So go ahead, ignore the GGU extension and implement as a dynamic configuration block.

Reply:
Don't forget to add an example in `example_of_dynamic_config_request.lua`.